### PR TITLE
DM-9022: Implement Flask skeleton

### PR DESCRIPTION
**Pull request by jonathansick**
This PR sets up the basic infrastructure for a Flask app. This is mostly ported from lsst-sqre/ltd-keeper.

### CODE REVIEW

**Comment by jonathansick** [comment]
That's right. It's meant to work inside the `lsst-the-docs` system, not `api.lsst.codes`.

File: app/testutils.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    commands and return a `response` `namedtuple` with fields:
+
+    - `status`: the integer HTTP response code
+    - `header`: the HTTP response headers
+    - `json`: the return data, parse as JSON into a Python `dict` object.
+    """
+    def __init__(self, app, username='', password=''):
+        self.app = app
+        self.auth = 'Basic ' + b64encode((username + ':' + password)
+                                         .encode('utf-8')).decode('utf-8')
```
Comment (athornton) [inline app/testutils.py]: Why are the encode/decode needed?  Just to force UTF-8 as the character set?

File: app/testutils.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        """Send a generate HTTP request."""
+        if headers is None:
+            headers = {}
+
+        # for testing, URLs just need to have the path and query string
+        url_parsed = urlsplit(url)
+        url = urlunsplit(('', '', url_parsed.path, url_parsed.query,
+                          url_parsed.fragment))
+
+        # append the autnentication headers to all requests
```
Comment (athornton) [inline app/testutils.py]: "authentication"

File: app/testutils.py (hunk 1 above)
Comment (jonathansick) [inline app/testutils.py]: Good question. That one comes from the O'Reilly book (https://github.com/miguelgrinberg/flasky/blob/master/tests/test_api.py#L27) and wasn't described.

**Pull request merged**